### PR TITLE
Test the voting pipeline

### PR DESCRIPTION
**Pull request by Arbaba**
This PR adds tests for the whole voting pipeline: contract deployment, administration and voting.

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -1,49 +1,268 @@
...
+      await waitFor(() => {
+        expect(Client.prototype.contractSubmissionPipeline).toHaveBeenCalledWith(
+          createContract(voters.length),
+          CONTRACTGAS,
+          initContractState(adminAddress, voters),
+          voters.length.toString()
+        )
+        expect(Client.prototype.contractSubmissionPipeline).toHaveBeenCalledTimes(1)
+        expect(Client.prototype.getTxStatus).toHaveBeenCalledTimes(1)
+        expect(screen.getByText('confirmed !')).toBeInTheDocument()
```
Comment (nop33) [inline src/App.test.tsx:119]: Is adding a space before punctuation a French thing? 😅

File: src/pages/Administrate.tsx [hunk 2]
```diff
@@ -81,8 +83,9 @@ const Administrate = () => {
       )}
       {!txResult && (
         <Container>
-          <p>Contract transaction ID</p>
+          <label htmlFor="tx-id">Contract transaction ID</label>
```
Comment (nop33) [inline src/pages/Administrate.tsx:86]: Good job with applying the best practices of testing-library and improving accessibility 🎖️

**Review by nop33 — COMMENTED** [review]
Really good job @Arbaba 👏

**Review by nop33 — APPROVED** [review]

File: src/App.test.tsx (hunk 1 above)
Comment (Arbaba) [reply src/App.test.tsx:119]: Yes it is 😅, will fix this

**Comment by Arbaba** [comment]
Thanks for the feedback @nop33!

**Pull request merged**